modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/portal/SiteHandler.java

```diff
@@ -32,6 +40,21 @@ protected PortalRequest createPortalRequest( final WebRequest webRequest, final
         return doCreatePortalRequest( webRequest, BASE_URI, baseSubPath );
     }
 
+    @Override
+    protected void doHandleBeforeNextInChain( final PortalRequest portalRequest )
+    {
+        if ( !"false".equalsIgnoreCase( System.getProperty( DISALLOW_ACCESS_TO_DRAFT_BRANCH_FOR_ANONYMOUS_CONFIG ) ) &&
+            ContentConstants.BRANCH_DRAFT.equals( portalRequest.getBranch() ) )
+        {
+            final AuthenticationInfo authInfo = ContextAccessor.current().getAuthInfo();
+            if ( !( authInfo.hasRole( RoleKeys.ADMIN ) || authInfo.hasRole( RoleKeys.ADMIN_LOGIN ) ) &&
```

Comment:
Anonymous should get Authentication Required 
Any other with insufficient rights should get Forbidden